### PR TITLE
Static analysis

### PR DESCRIPTION
**Pull request by SamWilsn**

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+on: [push, pull_request, workflow_dispatch]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python: [3.7, pypy-3.7]
```
Comment (lightclient) [inline .github/workflows/test.yaml:11]: why `3.7`? was thinking we were going for `3.9`?

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[build-system]
+requires = ["setuptools", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[tool.isort]
+profile = "black"
+multi_line_output = 3
+
+[tool.black]
+line-length = 79
```
Comment (lightclient) [inline pyproject.toml:10]: :rocket:

**Review by lightclient — APPROVED** [review]

File: .github/workflows/test.yaml (hunk 1 above)
Comment (SamWilsn) [reply .github/workflows/test.yaml:11]: That's what pypy supports, and we aren't using the annotated type stuff after all. And trust me, we very much want pypy if we want to get any reasonable performance.

Comment (lightclient) [reply .github/workflows/test.yaml:11]: sounds good :)

**Pull request merged**